packages/rum/test/benchmarks/run.js

```diff
@@ -46,31 +47,34 @@ const REPORTS_DIR = join(__dirname, '../../reports')
     await customApmBuild(filename)
 
     const server = await startServer(filename)
+    const close = promisify(server.close.bind(server))
     /**
      * object cache holding the metrics accumlated in each run and
      * helps in processing the overall results
      */
     const resultMap = new Map()
 
     for (let scenario of scenarios) {
-      const browser = await launchBrowser()
-      const url = `http://localhost:${port}/${scenario}`
-      const version = await browser.version()
-
-      /**
-       * Add common set of metrics for all scenarios
-       */
-      resultMap.set(scenario, getCommonFields({ version, url, scenario }))
-
-      for (let i = 0; i < runs; i++) {
-        const metrics = await gatherRawMetrics(browser, url)
-        Object.assign(metrics, { scenario, url })
-        await analyzeMetrics(metrics, resultMap)
+      for (let type of browserTypes) {
+        const url = `http://localhost:${port}/${scenario}`
+        /**
+         * Add common set of metrics for all scenarios
+         */
+        const key = `${scenario}.${type}`
+        resultMap.set(key, getCommonFields({ browser: type, url, scenario }))
+        for (let i = 0; i < runs; i++) {
+          const browser = await launchBrowser(type)
+          const metrics = await gatherRawMetrics(browser, url)
+          Object.assign(metrics, { scenario: key, url })
+          await analyzeMetrics(metrics, resultMap)
+          await browser.close()
+        }
```

Comment:
Isn't it better to first iterate over the browsers and then scenarios, e.g. run all scenarios on chrome, then on firefox etc.

Reply:
Reasoning behind is to keep scenario as the top level. scenario.browser.* like how the benchmark response is returned.